public/scripts/extensions/stable-diffusion/settings.html

```diff
@@ -189,7 +190,82 @@
                     </label>
                 </div>
             </div>
-
+            <div data-sd-source="stability">
+                <label for="sd_stability_key">API Key</label>
+                <div class="flex-container flexnowrap">
+                    <input id="sd_stability_key" type="password" class="text_pole flex1" placeholder="Enter your Stability AI API key" />
+                    <div id="sd_stability_validate" class="menu_button menu_button_icon">
+                        <i class="fa-solid fa-check"></i>
+                        <span data-i18n="Connect">
+                            Connect
+                        </span>
+                    </div>
+                </div>
+                <i>You can find your API key in the Stability AI dashboard.</i>
+            
+                <div class="flex-container">
+                    <div class="flex1">
+                        <label for="sd_stability_engine">Engine</label>
+                        <select id="sd_stability_engine">
+                            <option value="v2beta">V2beta Image Generation</option>
+                        </select>
+                    </div>
+                    <div class="flex1">
+                        <label for="sd_stability_style_preset">Style Preset</label>
+                        <select id="sd_stability_style_preset">
+                            <option value="anime">Anime</option>
+                            <option value="3d-model">3D Model</option>
+                            <option value="analog-film">Analog Film</option>
+                            <option value="cinematic">Cinematic</option>
+                            <option value="comic-book">Comic Book</option>
+                            <option value="digital-art">Digital Art</option>
+                            <option value="enhance">Enhance</option>
+                            <option value="fantasy-art">Fantasy Art</option>
+                            <option value="isometric">Isometric</option>
+                            <option value="line-art">Line Art</option>
+                            <option value="low-poly">Low Poly</option>
+                            <option value="modeling-compound">Modeling Compound</option>
+                            <option value="neon-punk">Neon Punk</option>
+                            <option value="origami">Origami</option>
+                            <option value="photographic">Photographic</option>
+                            <option value="pixel-art">Pixel Art</option>
+                            <option value="tile-texture">Tile Texture</option>
+                        </select>
+                    </div>
+                </div>
+            
+                <div class="flex-container">
+                    <div class="flex1">
+                        <label for="sd_stability_aspect_ratio">Aspect Ratio</label>
```

Comment:
Is this needed? We got a selector for width/height below everything else.

Reply:
Yes, new stable image models do not use height or width—they use aspect ratio
https://platform.stability.ai/docs/api-reference#tag/Generate/paths/~1v2beta~1stable-image~1generate~1ultra/post

![Screenshot_20240704-075155](https://github.com/SillyTavern/SillyTavern/assets/38446123/c65fcace-7595-4194-838b-1b6bed758224)

Reply:
We could normalize the aspect ratio from selected width and height by selecting the closest aspect ratio to the list as we do for OpenAI image generation (it only allows 4  specific sizes). If you don't feel like it, I can take this task on me.

Reply:
please do!!!